docs/distribution/README.md

```diff
@@ -0,0 +1,528 @@
+
+### Signature Persistance - Option 1: oci-manifest
+
+The challenge with using oci-manifest is how the registry tracks the linkage between the signature and the original artifact.
+
+<img src="../../media/signature-as-manifest.png" width=400>
+
+Example **manifest** for a Notary v2 signature
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
+    "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "size": 1906
+  },
+  "layers": []
+}
+```
+
+**Pros with this approach:**
+
+* OCI Artifacts already supports manifest based artifacts, through the `manifest.config.mediaType`
+
+**Cons with this approach:**
+
+* Manifests have no means to reference other artifacts.
+* An alternative is required to link a target artifact with it's signature. Either through parsing the signature `manifest.config` object, or [a separate API for linking objects](#linking-signatures-to-artifacts).
+
+### Signature Persistance - Option 2: oci-index
+
+This option is similar to using oci-manifest. However, instead of parsing the signature object to determine the linkage between an artifact and signature, the `index.manifests` collection is utilized.
+
+<img src="../../media/signature-as-index.png" width=400>
+
+Example **index** for a Notary v2 signature
+
+``` json
+{
+  "schemaVersion": 2.1,
+  "mediaType": "application/vnd.oci.image.index.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
+    "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "size": 1906
+  },
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+      "size": 7023,
+      "platform": {
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+**Pros with this approach:**
+
+* Utilize the existing `index.manifests` collection for linking artifacts
+* Registries that support oci index already have infrastructure for tracking `index.manifests`, including delete operations and garbage collection
+* Existing distribution-spec upload APIs are utilized
+* Unlike the manifest proposal, no additional artifact handler would be required to parse the config object for linking artifacts
+* Based on the artifact type:  `manifest.config.mediaType: "application/vnd.cncf.notary.config.v2+jwt"`, role check may be done to confirm the identity has a signer role
+* As registry operators may offer role checking for different artifact types, Notary v2 Signatures are just one of many types they may want to authorize
+
+**Cons with this approach:**
+
+* OCI index does not yet support the [OCI config descriptor][oci-descriptor]. This would require a schema change to oci-index, with a version bump.
```

Comment:
We need a place to store the signature.
Since we need the `config.mediaType` to declare the artifact type, it seemed like a convienant place to store the signature.
- Option 2 stores the signature in the config object
- Option 3 stores the signature in another manifest

The proposal mentioned is essentially option 3.